comixed-rest-api/src/main/java/org/comixedproject/messaging/ComiXedWebSocketChannelInterceptor.java

```diff
@@ -0,0 +1,82 @@
+ * Copyright (C) 2021, The ComiXed Project
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses>
+ */
+
+package org.comixedproject.messaging;
+
+import java.util.List;
+import lombok.extern.log4j.Log4j2;
+import org.comixedproject.authentication.ComiXedUserDetailsService;
+import org.comixedproject.authentication.JwtTokenUtil;
+import org.comixedproject.messaging.model.StompPrincipal;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.simp.stomp.StompCommand;
+import org.springframework.messaging.simp.stomp.StompHeaderAccessor;
+import org.springframework.messaging.support.ChannelInterceptor;
+import org.springframework.messaging.support.MessageHeaderAccessor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+/**
+ * <code>ComiXedWebSocketChannelInterceptor</code> provides a security interceptor to create the
+ * principal for the messaging layer.
+ *
+ * @author Darryl L. Pierce
+ */
+@Component
+@Log4j2
+public class ComiXedWebSocketChannelInterceptor implements ChannelInterceptor {
+  private static final String HEADER_STRING = "Authorization";
+
+  @Autowired private ComiXedUserDetailsService userDetailsService;
+  @Autowired private JwtTokenUtil jwtTokenUtil;
+
+  @Override
+  public Message<?> preSend(final Message<?> message, final MessageChannel channel) {
+    StompHeaderAccessor accessor =
+        MessageHeaderAccessor.getAccessor(message, StompHeaderAccessor.class);
+    if (accessor != null && StompCommand.CONNECT.equals(accessor.getCommand())) {
+      final List<String> authorization = accessor.getNativeHeader(HEADER_STRING);
+      if (authorization != null && !authorization.isEmpty()) {
+        log.trace("Auth token found");
+        final String token = authorization.get(0);
+        doCreatePrincipal(accessor, token);
+      }
+    }
+    return message;
+  }
+
+  void doCreatePrincipal(final StompHeaderAccessor accessor, final String token) {
+    String username = null;
+    try {
+      username = this.jwtTokenUtil.getEmailFromToken(token);
+    } catch (Exception error) {
+      log.error("Unable to extract username from auth token", error);
+    }
+    if (username != null) {
```

Comment:
Shouldn't we check if the username is blank?

Reply:
Yeah, I suppose we should. Normally what should come out of the encoded token is either a value or a null. But to be well-balanced, we can use StringUtils.isEmpty() to be complete.

Reply:
I've also applied these changes to ComiXedAuthenticationFilter since it's doing similar behavior.